kratos/utilities/parallel_levelset_distance_calculator.h

```diff
@@ -148,153 +151,116 @@ class ParallelDistanceCalculator
-
-                if(IsActive(visited))
-                {
-                    double Volume;
-                    GeometryUtils::CalculateGeometryData(geom,DN_DX,N,Volume);
-
-                    AddDistanceToNodes(rDistanceVar,rAreaVar,geom,DN_DX,Volume);
+            block_for_each(rModelPart.Elements(), tls_container, [&](Element& rElement, TLSType& rTLSContainer){
+                auto& r_geom = rElement.GetGeometry();
+                auto& r_visited = std::get<0>(rTLSContainer);
+                auto& r_N = std::get<1>(rTLSContainer);
+                auto& r_DN_DX = std::get<2>(rTLSContainer);
+                for (unsigned int j=0; j<TDim+1; j++) {
+                    r_visited[j] = r_geom[j].Is(VISITED);
                 }
-            }
+                if (IsActive(r_visited)) {
+                    double volume;
+                    GeometryUtils::CalculateGeometryData(r_geom, r_DN_DX, r_N, volume);
+                    AddDistanceToNodes(rDistanceVar, rAreaVar, r_geom, r_DN_DX, volume);
+                }
+            });
 
             //mpi sync variables
             if(is_distributed == true)
             {
-                #pragma omp parallel for private(DN_DX)
-                for(int i = 0; i<node_size; i++)
-                {
-                    ModelPart::NodesContainerType::iterator it=rModelPart.NodesBegin()+i;
-                    if(it->GetValue(IS_VISITED) == 1.0)
-                    {
-                        double& distance = it->FastGetSolutionStepValue(rDistanceVar);
-                        it->GetValue(rDistanceVar) = distance;
-                        distance = 0.0;
+                block_for_each(rModelPart.Nodes(), [&](NodeType& rNode){
+                    if (rNode.Is(VISITED)) {
+                        double& r_distance = rNode.FastGetSolutionStepValue(rDistanceVar);
+                        rNode.GetValue(rDistanceVar) = r_distance;
+                        r_distance = 0.0;
+                    } else {
+                        rNode.GetValue(rDistanceVar) = 0.0;
                     }
-                    else
-                        it->GetValue(rDistanceVar) = 0.0;
-                }
+                });
 
                 rModelPart.GetCommunicator().AssembleCurrentData(rAreaVar);
                 rModelPart.GetCommunicator().AssembleCurrentData(rDistanceVar);
 
-                #pragma omp parallel for private(DN_DX)
-                for(int i = 0; i<node_size; i++)
-                {
-                    ModelPart::NodesContainerType::iterator it=rModelPart.NodesBegin()+i;
-                    it->FastGetSolutionStepValue(rDistanceVar) += it->GetValue(rDistanceVar);
-                }
+                block_for_each(rModelPart.Nodes(), [&](NodeType& rNode){
+                    rNode.FastGetSolutionStepValue(rDistanceVar) += rNode.GetValue(rDistanceVar);
+                });
 
                 rModelPart.GetCommunicator().GetDataCommunicator().Barrier();
```

Comment:
```suggestion
```

I don't see the reason for this, can you have a look?
IMO this is not necessary, esp since there are assemble calls right before

Reply:
Let me leave this for an upcoming PR. I agree it seems to not be needed.